src/opentimelineio/stackAlgorithm.cpp

```diff
@@ -123,10 +125,49 @@ _flatten_next_item(
     }
 }
 
+// make all tracks the same length by adding a gap to the end if they are shorter
```

Comment:
Could you add a comment clarifying that this updates the `tracks` vector with a mix of old unchanged tracks and new padded tracks? At first read I was confused by that.

Reply:
Also, do you think anyone else might want to call this function for other use cases? I could imagine that some adapters might want to have equal-duration padded tracks. We could leave that for a future PR.

Reply:
It could be useful, not sure. I can't think of anywhere atm in the AAF adapter this would be needed. I'm not sure if any other adapters.